wagtail/models/sites.py

```diff
@@ -68,12 +114,38 @@ def get_queryset(self):
     def get_by_natural_key(self, hostname, port):
         return self.get(hostname=hostname, port=port)
 
+    def get_all(self) -> Tuple["Site"]:
```

Comment:
Question: Why is this a tuple? It ought to be a list.

Reply:
I found some information in the `weakref` [module docs](https://docs.python.org/3/library/weakref.html), which I think is relevant here:

> Several built-in types such as [list](https://docs.python.org/3/library/stdtypes.html#list) and [dict](https://docs.python.org/3/library/stdtypes.html#dict) do not directly support weak references but can add support through subclassing:

```
class Dict(dict):
    pass

obj = Dict(red=1, green=2, blue=3)   # this object is weak referenceable
```

> CPython implementation detail: Other built-in types such as [tuple](https://docs.python.org/3/library/stdtypes.html#tuple) and [int](https://docs.python.org/3/library/functions.html#int) do not support weak references even when subclassed.

Since [`asgiref.local` makes use of `threading.local`](https://github.com/django/asgiref/blob/main/asgiref/local.py#L76) and [`threading.local` makes use of `weakref`](https://github.com/python/cpython/blob/3.12/Lib/_threading_local.py#L183-L184), it feels like the safest and lowest complexity option here is to define and use custom list types, which I've now done.